src/doc/unstable-book/src/language-features/register-tool.md

```diff
@@ -0,0 +1,48 @@
+# `register_tool`
+
+The tracking issue for this feature is: [#66079]
+
+[#66079]: https://github.com/rust-lang/rust/issues/66079
+
+------------------------
+
+The `register_tool` language feature informs the compiler that attributes in your code are meant to be used with tools other than the compiler itself. This can be useful if your code has semantic meaning without the external tool, but enables additional features when the tool is present.
+
+`register_tool` also allows configuring lint levels for external tools.
+
+Tool attributes are only meant for ignorable attributes. If your code *changes* meaning when the attribute is present, it should not use a tool attribute (because it cannot be compiled with anything other than the external tool, and in a sense is a fork of the language).
+
+------------------------
+
+`#![register_tool(tool)]` is an attribute, and is only valid at the crate root.
+Attributes using the registered tool are checked for valid syntax, and lint attributes are checked to be in a valid format. However, the compiler cannot validate the semantics of the attribute, nor can it tell whether the configured lint is present in the external tool.
+
+Semantically, `clippy::*`, `rustdoc::*`, and `rustfmt::*` lints and attributes all behave as if `#![register_tool(clippy, rustdoc, rustfmt)]` were injected into the crate root, except that the `rustdoc` namespace can only be used for lints, not for attributes.
+When compiling with `-Z unstable-features`, `rustc::*` lints can also be used. Like `rustdoc`, the `rustc` namespace can only be used with lints, not attributes.
+
+The compiler will emit an error if it encounters a lint/attribute whose namespace isn't a registered tool.
+
+Tool names cannot be "nested"; `register_tool(foo::bar)` is an error.
+
+## Examples
+
+Tool attributes:
+
+```rust
+#![feature(register_tool)]
+#![register_tool(c2rust)]
+
+#[c2rust::header_src = "/home/user/some/workspace/foobar/bar.h:5"]
+pub mod bar_h {
```

Comment:
@kkysen are you ok with me mentioning c2rust in the official docs here?

Reply:
You can also just use some kind fo stub tool name 🤷

Reply:
Sure, I think so. @thedataking, this is fine, right?

Reply:
> You can also just use some kind of stub tool name 🤷

i think giving an example that shows how the attribute is *useful* is better than just shoving in foo/bar everywhere. motivating examples, and all that.

(of course it doesn't need to be c2rust specifically; i'm happy to find some other external tool if Per doesn't want it mentioned here.)

Reply:
For what it's worth, I think it'd be very cool to mention c2rust here.

Reply:
> Sure, I think so. @thedataking, this is fine, right?
Definitely a good idea IMHO. Please go ahead @jyn514.